src/rust/ensogl/src/display/shape/primitive/system.rs

```diff
@@ -58,7 +58,7 @@ impl ShapeSystem {
         material.add_input  ("time"         , 0.0);
         material.add_input  ("symbol_id"    , 0);
         material.add_input  ("display_mode" , 0);
-        material.add_output ("id"           , Vector4::<u32>::new(0,0,0,0));
+        material.add_output ("id"           , Vector4::<f32>::new(0.0,0.0,0.0,0.0));
```

Comment:
I know it was already ugly, but we can do better now. You can use `::zero()` instead :)